spring-pulsar/src/main/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainer.java

```diff
@@ -296,71 +310,83 @@ public void run() {
 				// Always receive messages in batch mode.
 				try {
 					if (!inRetryMode.get() && !messagesPendingInBatch.get()) {
+						DefaultPulsarMessageListenerContainer.this.receiveInProgress = true;
 						messages = this.consumer.batchReceive();
+						DefaultPulsarMessageListenerContainer.this.receiveInProgress = false;
```

Comment:
If `batchReceive` fails for some other reason we do not clear the flag. I think we should move the set(false) to finally block rather than directly below the `batchReceive`.